paimon-common/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -3226,4 +3311,31 @@ public String toString() {
             return value;
         }
     }
+
+    /** Actions performed during table writing. */
+    public enum WriteAction {
+
+        // All write actions will be performed, this is the default behavior.
+        ALL("all"),
+
+        // Actions during commit.
+        PARTITION_EXPIRE("partition-expire"),
+        SNAPSHOT_EXPIRE("snapshot-expire"),
+        TAG_AUTOMATIC_CREATION("tag-automatic-creation"),
+
+        // Actions during writing.
+        MINOR_COMPACT("minor-compact"),
```

Comment:
It is strange here, in streaming mode, it is hard to say just doing minor compact.

Reply:
We can only keep `full-compact` and `minor-compact` action can be removed.